WordPress/WordPressTest/Mention/SuggestionsTableViewTests.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  SuggestionsTableViewTests.swift
+//  WordPressTest
+//
+//  Created by Salim Braksa on 8/7/2022.
+//  Copyright © 2022 WordPress. All rights reserved.
+//
+
+import XCTest
+
+@testable import WordPress
+
+class SuggestionsTableViewTests: CoreDataTestCase {
+
+    private var suggestionsTableView: SuggestionsTableView!
+    private var delegate: SuggestionsTableViewDelegate!
+    private var suggestions: [UserSuggestion] = []
+
+    // MARK: - Lifecycle
+
+    override func setUpWithError() throws {
+        self.suggestions = try self.loadUserSuggestions()
+        self.delegate = SuggestionsTableViewMockDelegate()
+        self.suggestionsTableView = SuggestionsTableView(siteID: 1, suggestionType: .mention, delegate: delegate)
+        self.suggestionsTableView.suggestions = suggestions
+    }
+
+    override func tearDownWithError() throws {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+    }
+
+    // MARK: - Tests
+
+    /// Test to make sure that `suggestions` are loaded.
+    func testSuggestionsCount() {
+        XCTAssertEqual(suggestions.count, 100)
+    }
+
+    func test_moveProminentSuggestionsToTop() {
+        let input = suggestions
+
+        // Case #1
+        let expectedResult1 = input
+        let result1 = self.suggestionsTableView.moveProminentSuggestionsToTop(searchResults: input, prominentSuggestionsIds: [])
+
+        // Case #2
+        var expectedResult2 = input
+        var element = expectedResult2.remove(at: 49)
+        expectedResult2.insert(element, at: 0)
+        let result2 = self.suggestionsTableView.moveProminentSuggestionsToTop(searchResults: input, prominentSuggestionsIds: [50])
+
+        // Case #3
+        var expectedResult3 = input
+        element = expectedResult3.remove(at: 9)
+        expectedResult3.insert(element, at: 0)
+        element = expectedResult3.remove(at: 14)
+        expectedResult3.insert(element, at: 1)
+        let result3 = self.suggestionsTableView.moveProminentSuggestionsToTop(searchResults: input, prominentSuggestionsIds: [10, 15])
```

Comment:
@twstokes Is it okey to construct these `expectedResults` like this? Another alternative is to have those expectations as JSON files, then load them the same way we are loading `user-suggestions.json`.

Reply:
It's OK, but it may be even clearer to split up those cases into their own tests. I think tests are a place where we can be more verbose and not as focused on being DRY.

I've personally been a fan of the "[given / when / then](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/WordPressTest/SiteCreation/SiteIntentDataTests.swift)" format.